### DISCUSSION TITLE
Code highlighting would be nice

### DISCUSSION THREAD

**Discussion by timlrx** [timlrx/tailwind-nextjs-starter-blog#78] [Q&A] [answered]
Something like [gatsby-remark-prism](https://www.gatsbyjs.com/plugins/gatsby-remark-prismjs/) syntax highlighting would be useful.

Code would be highlighted based on information passed in the markdown code cell e.g.

js{1,3-5} -> highlight line 1 and 3 to 5

It should function as a remark / rehype plugin possibly extending on [rehype-prism](https://github.com/mapbox/rehype-prism)

**Comment by Music47ell** [comment]
I tried [this one](https://www.npmjs.com/package/@deckdeckgo/highlight-code) with Gatsby and it looks great. [(GitHub repo)](https://github.com/deckgo/gatsby-remark-highlight-code).

**Reply by timlrx** [reply]
Looks, nice, unfortunately like prism.js this does client side parsing. We are currently doing everything on the server side so only HTML and css is generated. I believe the logic from either reference libraries should be relatively straightforward to copy over and just add a class name to highlighted lines

**Reply by Music47ell** [reply]
I currently have this implemented locally. Different npm module, but the same result you want. 

I should have a PR ready soon.

**Reply by timlrx** [reply]
I actually wrote my own plugin - https://github.com/timlrx/rehype-prism-plus, publishing it tomorrow on npm. Sorry for not informing earlier!

**Reply by Music47ell** [reply]
My implementation uses [prism-react-renderer](https://github.com/FormidableLabs/prism-react-renderer).

It comes bundled with many [themes](https://github.com/FormidableLabs/prism-react-renderer/tree/master/src/themes) and it doesn't require rehype or remark.

Here's a screenshot of the final result:

<img width="791" alt="Screen Shot 2021-06-28 at 6 50 25 PM" src="https://user-images.githubusercontent.com/5708563/123666375-d13f6a00-d841-11eb-9795-cd8e79231d19.png">

**Reply by timlrx** [reply]
That is another client-side approach. From bundlephobia, it adds another 22kb of js which is half the size of the current bundle. Hence, my preference for a server-side approach where we generate all the necessary mark-up at build time. With the new package I have written, it should be almost feature parity with support for 258 languages, no additional imports required. Thought this comes with additional work of configuring a stylesheet rather than having themes.

**Reply by Music47ell** [reply]
Ok, that sounds great. Will it be possible to use Tailwind for styling? 
When should we expect this to be implemented?

**Reply by timlrx** [reply]
Yes, it uses tailwind for styling! It's available on the mdx-bundler branch: https://github.com/timlrx/tailwind-nextjs-starter-blog/tree/mdx-bundler. Updated the v1 blog post with some information as well.

There's still some work to be done to unlock the full power of the new xdm mdx compiler and update some of the blog post to deal with the differences in the compilers. Once those are done, I will merge it over to v1. I also want to do a simple benchmark of the two compilers before switching over,

**Reply by timlrx** [reply]
Generated a preview PR and here's the section of interest - https://tailwind-nextjs-starter-blog-ahdkf4lvx-timlrx.vercel.app/blog/new-features-in-v1#line-highlighting-and-line-numbers

**Reply by Music47ell** [reply]
This looks pretty good. I've been hearing a lot of good things about mdx-bundler. Glad to see it implemented here. 
I can't wait to see the benchmark.

**Comment by timlrx** [comment, marked as answer]
Implemented with [rehype-prism-plus](https://github.com/timlrx/rehype-prism-plus). Here's a [preview link](https://tailwind-nextjs-starter-blog-ahdkf4lvx-timlrx.vercel.app/blog/new-features-in-v1#line-highlighting-and-line-numbers) while it is pending merging.
